### ISSUE TITLE
controller.h

### ISSUE THREAD

**Issue by madscientist64** [BigJBehr/ESP32-Bluetooth-BLE-Remote-Control#2]
sorry, maybe a bit of a noob question but i'm trying to compile the code and i'm getting this on my Arduino IDE: 

----------------
Gatt-VRBOX-Tasks:42:24: error: controller.h: No such file or directory
compilation terminated.
exit status 1
controller.h: No such file or directory
----------------

am I missing something? is it possible I just need to update my esp32 Arduino IDE libraries?

**Comment by GiacomoGenovese** [comment]
Hi,
I have the same problem.

**Comment by BigJBehr** [comment]
As I have stated before, I am sorry you are having issues. The problem is that in the time since I first wrote and release this software, Expressif changed the Arduino SDK for the ESP32. The current SDK is not compatible with my code. Expressif decided to change the number of parameters on some functions and eliminated several .h files. This is why you are having these problems. I have tried to fix the issues, but ran into a wall due to an error that Expressif introduced into their Bluetooth stack that causes the stack to crash if more than one characteristic of a service is opened. Expressif has to fix this, I cannot.

**Comment by GiacomoGenovese** [comment]
Thanks for reply.

**Comment by GiacomoGenovese** [comment]
I moved to a BT 3.0 device, AB shutter 3, I will try to connect it to the ESP32 and control light by the button in the shutter. Any help is welcome. :)
Regards

**Comment by pipercolt** [comment]
any update on this , was thinking of using bigbehrs joystick code .

**Comment by BigJBehr** [comment]
I have completely rewritten this project. The new code is based on Neil Kolbane's Arduino BLE framework. It works without any problems. I will post an update with the new code as soon as I get a chance.

**Comment by BigJBehr** [comment]
I have added the file BLE-VRBOX.ino to the repository. This is the only file that you need. Compile it with Arduino ver 1.8.12 with the ESP32 addon from Expressif.  You will need to add code for the action that you want to take when a button is pressed or the joystick moves. Not all the button work in the same way. Be careful with the C & D buttons. The VRBOX device must be in Mouse mode (@ + D button). Make sure that the VRBOX is turned on before starting the ESP32 or it may not find the VRBOX.

**Comment by pipercolt** [comment]
Get an error undefined reference to `BLERemoteService::getCharacteristicsByHandle() on compilation

**Comment by BigJBehr** [comment]
The error is caused by the wrong ESP32 BLE library. I have updated the readme.md file to include a link to the repository for Neil Kolban's ESP32 BLE library. Download as a zip file and the use the Arduino's Library Manager to install the zip library. This should solve all your problems.

**Comment by pipercolt** [comment]
hey Behr,  I've downloaded the zip file from neil's library as stated and installed it via the Arduino add zip library.   same error.   I tried on a second machine with a new install of the IDE , same result.  
undefined reference to `BLERemoteService::getCharacteristicsByHandle()  
when I navigate to BLERemoteService.h in libraries its defined as
	std::map<uint16_t, BLERemoteCharacteristic*>* getCharacteristicsByHandle();  // Get the characteristics map.
I'm not a big template guy so not sure how this is mapped.
